src/test/java/QueryInformationTest.java

```diff
@@ -0,0 +1,146 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.sps.data.QueryInformation;
+import java.text.ParseException;
+import java.util.TimeZone;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public final class QueryInformationTest {
+  private static final String EMPTY_STRING = "";
+  private static final String NULL_VALUE = null;
+  private static final ImmutableSet<String> EMPTY_IMMUTABLE_SET = ImmutableSet.of();
```

Comment:
nitnit: not sure these are necessary. You could probably reference them directly in the test.